src/main/resources/org/cbioportal/legacy/persistence/mybatis/StudyMapper.xml

```diff
@@ -178,4 +178,98 @@
         </where>
     </select>
 
+    <select id="getResourceCountsForAllStudies" resultType="org.cbioportal.legacy.model.ResourceCount">
+        SELECT
+            rs.RESOURCE_ID AS resourceId,
+            DISPLAY_NAME AS displayName,
+            rd.DESCRIPTION AS description,
+            rd.RESOURCE_TYPE AS resourceType,
+            rd.PRIORITY AS priority,
+            rd.OPEN_BY_DEFAULT AS openByDefault,
+            CANCER_STUDY_IDENTIFIER as cancerStudyIdentifier,
+            rd.CUSTOM_METADATA AS customMetaData,
+            COUNT(DISTINCT rs.INTERNAL_ID) AS sampleCount,
+            COUNT(DISTINCT p.INTERNAL_ID) AS patientCount
+        FROM resource_sample AS rs
+                 INNER JOIN resource_definition rd ON rd.RESOURCE_ID = rs.RESOURCE_ID
+                 INNER JOIN sample s ON s.INTERNAL_ID = rs.INTERNAL_ID
+                 INNER JOIN patient p ON p.INTERNAL_ID = s.PATIENT_ID
+                 INNER JOIN cancer_study cs ON cs.CANCER_STUDY_ID = p.CANCER_STUDY_ID
```

Comment:
There are some shared code blocks between these two methods, I think we can reuse some of the code by creating a shared sql to be referenced.
You can see reference implementation here.
https://github.com/cBioPortal/cbioportal/blob/master/src/main/resources/org/cbioportal/legacy/persistence/mybatis/GeneMapper.xml#L6-L20